chore: fix C lint errors (issue #8443)

kaushal-kumar-it opened the pull request:
Resolves #8443.

## Description

> What is the purpose of this pull request?

This pull request:

- Fixes C linting errors in the BLAS `dasum` benchmark files.
- Replaces uninitialized variable usage and VLA arrays with properly allocated arrays using `malloc`.
- Removes unnecessary `malloc` null checks to comply with stdlib benchmark style.
- Ensures all benchmark arrays are initialized and freed correctly to avoid warnings and memory leaks.
- Improves code consistency with other stdlib benchmark files.

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

- #8443

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

- [ ] Yes
- [x] No

If you answered "yes" above, how did you use AI assistance?

- [ ] Code generation (e.g., when writing an implementation or fixing a bug)
- [ ] Test/benchmark generation
- [ ] Documentation (including examples)
- [ ] Research and understanding

#### Disclosure

> If you answered "yes" to using AI assistance, please provide a short disclosure indicating how you used AI assistance. This helps reviewers determine how much scrutiny to apply when reviewing your contribution.

N/A

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

kaushal-kumar-it committed:
chore: fix C lint errors (issue #8443)

kaushal-kumar-it committed:
chore: fix C lint errors (issue #8443)

kgryte commented on lib/node_modules/@stdlib/blas/base/dasum/benchmark/c/benchmark.length.c:
```diff
@@ -74,7 +74,7 @@ static void print_results( int iterations, double elapsed ) {
 	struct timeval now;
 	gettimeofday( &now, NULL );
-	return (double)now.tv_sec + (double)now.tv_usec/1.0e6;
+	return (double)now.tv_sec + (double)now.tv_usec / 1.0e6;
```
```suggestion
	return (double)now.tv_sec + (double)now.tv_usec/1.0e6;
```

kgryte committed:
style: visually group operands

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dasum/benchmark/c/benchmark.length.c:
```diff
@@ -171,7 +173,7 @@ int main( void ) {
 	for ( i = MIN; i <= MAX; i++ ) {
 		len = pow( 10, i );
-		iter = ITERATIONS / pow( 10, i-1 );
+		iter = ITERATIONS / pow( 10, i - 1 );
```
```suggestion
		iter = ITERATIONS / pow( 10, i-1 );
```

kgryte committed:
style: remove whitespace

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dasum/benchmark/c/benchmark.length.c:
```diff
@@ -188,4 +190,5 @@ int main( void ) {
 		}
 	}
 	print_summary( count, count );
+
```
```suggestion
```

kgryte committed:
style: remove empty line

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dasum/benchmark/c/benchmark.length.c:
```diff
@@ -129,13 +130,13 @@ static double benchmark1( int iterations, int len ) {
 
 	for ( i = 0; i < len; i++ ) {
-		x[ i ] = ( rand_double()*20000.0 ) - 10000.0;
+		x[ i ] = ( rand_double() * 20000.0 ) - 10000.0;
```
```suggestion
		x[ i ] = ( rand_double()*20000.0 ) - 10000.0;
```

kgryte committed:
style: remove whitespace

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dasum/benchmark/c/benchmark.length.c:
```diff
@@ -96,13 +96,13 @@ static double rand_double( void ) {
 
 	for ( i = 0; i < len; i++ ) {
-		x[ i ] = ( rand_double()*20000.0 ) - 10000.0;
+		x[ i ] = ( rand_double() * 20000.0 ) - 10000.0;
```
```suggestion
		x[ i ] = ( rand_double()*20000.0 ) - 10000.0;
```

kgryte committed:
style: remove whitespace

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
style: move allocation to separate line

This ensures consistency with https://github.com/stdlib-js/stdlib/blob/99e8f1412d6a27eb27e5d0d02d9f81945d14fbd5/tools/snippets/benchmark/c/native/benchmark.length.c.

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
cc @GeoDaoyu as you've made a number of PRs for similar lint errors. Going forward, we'd like to move to dynamic allocation rather than disabling the lint rule.
